File: example/typescript/OtherModule/Web/Dtos.ts
namespace OtherModule.Web {
    export class OtherClassDto {
        id = STRING
        amount = NUMBER

        static toApi(dto: OtherClassDto): OtherClass {
            return new OtherClass(
                new OtherId(dto.id),
                dto.amount,
            )
        }

        static fromApi(api: OtherClass): OtherClassDto {
            const dto = new OtherClassDto()
            dto.id = api.id.value
            dto.amount = api.amount
            return dto
        }
    }
}

File: example/kotlin/src/main/kotlin/com/some/path/othermodule/web/Dtos.kt
package com.some.path.othermodule.web

import com.some.path.othermodule.api.OtherClass
import com.some.path.othermodule.api.OtherId
import pl.bratek20.othermodule.api.*

data class OtherClassDto(
    val id: String,
    val amount: Int,
) {
    fun toApi(): OtherClass {
        return OtherClass(
            id = OtherId(id),
            amount = amount,
        )
    }

    companion object {
        fun fromApi(api: OtherClass): OtherClassDto {
            return OtherClassDto(
                id = api.id.value,
                amount = api.amount,
            )
        }
    }
}